java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```
@@ -192,110 +188,129 @@ private static byte[] buildMessage(
+      } else {
+        return new WasmIsEnabledResponse(response.impressionData(), null);
+      }
+    } catch (RuntimeException e) {
+      log.warn("Could not check if toggle is enabled: {}", e.getMessage(), e);
+      return new WasmIsEnabledResponse(false, null);
     }
   }
 
-  public WasmResponse<VariantDef> getVariant(String toggleName, Context context)
+  public WasmVariantResponse getVariant(String toggleName, Context context)
       throws YggdrasilInvalidInputException {
-    Map<String, Boolean> strategyResults = customStrategiesEvaluator.eval(toggleName, context);
-    byte[] contextBytes = buildMessage(toggleName, context, strategyResults);
-
-    Variant variant = nativeInterface.checkVariant(enginePointer, contextBytes);
-    if (variant.name() != null) {
-      Payload payload = null;
-
-      VariantPayload variantPayload = variant.payload();
-
-      if (variantPayload != null) {
-        payload = new Payload();
-        payload.setType(variant.payload().payloadType());
-        payload.setValue(variant.payload().value());
-      }
-
-      if (variant.error() != null) {
-        String error = variant.error();
-        throw new YggdrasilInvalidInputException(error);
+    try {
+      Map<String, Boolean> strategyResults = customStrategiesEvaluator.eval(toggleName, context);
+      byte[] contextBytes = buildMessage(toggleName, context, strategyResults);
+
+      Variant variant = nativeInterface.checkVariant(enginePointer, contextBytes);
+      if (variant.name() != null) {
+        Payload payload = null;
+
+        VariantPayload variantPayload = variant.payload();
+
+        if (variantPayload != null) {
+          payload = new Payload();
+          payload.setType(variant.payload().payloadType());
+          payload.setValue(variant.payload().value());
+        }
+
+        if (variant.error() != null) {
+          String error = variant.error();
+          throw new YggdrasilInvalidInputException(error);
+        }
```

This check could be moved to just after the line 254, so JVM won't have to unnecessarily assign few bytes of memory in case of error, and to align with "fail fast" approach.